Add WebAssembly (runtime) startup callbacks (#31852)
dotnet/AspNetCore.Docs: aspnetcore/blazor/fundamentals/startup.md

  * In a [`BlazorWebView`](/mobile-blazor-bindings/walkthroughs/hybrid-hello-world#mainrazor-native-ui-page), no options are passed.
* `afterStarted(blazor)`: Called after Blazor is ready to receive calls from JS. For example, `afterStarted` is used to initialize libraries by making JS interop calls and registering custom elements. The Blazor instance is passed to `afterStarted` as an argument (`blazor`).

:::moniker-end

:::moniker range=">= aspnetcore-8.0"

Additional .NET WebAssembly runtime callbacks:

> * `onRuntimeConfigLoaded(config)`: Called when the boot configuration is downloaded. Allows the app to modify parameters (config) before the runtime starts:

Reviewer: ```suggestion
* `onRuntimeConfigLoaded(config)`: Called when the boot configuration is downloaded. Allows the app to modify parameters (config) before the runtime starts (the parameter is `MonoConfig` from [dotnet.d.ts](https://github.com/dotnet/runtime/blob/main/src/mono/browser/runtime/dotnet.d.ts)):
```

  ```javascript

>   const params = new URLSearchParams(location.search);
> 
>   export function onRuntimeConfigLoaded(config) {
>     config.environmentVariables["LIBRARY_INITIALIZER_TEST"] = "1";
> 
>     if (params.get("throwError") === "true") {
>         throw new Error("Error thrown from library initializer");
>     }
>   }

Reviewer: ```suggestion
  export function onRuntimeConfigLoaded(config) {
    // Sample: Enable startup diagnostic logging when the URL contains parameter debug=1
    const params = new URLSearchParams(location.search);
    if (params.get("debug") == "1") {
      config.diagnosticTracing = true;
    }
  }
```

  ```

> * `onRuntimeReady({ getAssemblyExports, getConfig })`: Called after the .NET WebAssembly runtime has started:

Reviewer: ```suggestion
* `onRuntimeReady({ getAssemblyExports, getConfig })`: Called after the .NET WebAssembly runtime has started (the parameter is `RuntimeAPI` from [dotnet.d.ts](https://github.com/dotnet/runtime/blob/main/src/mono/browser/runtime/dotnet.d.ts)):
```

  ```javascript

>   export async function onRuntimeReady({ getAssemblyExports, getConfig }) {
>     const testCase = params.get("test");
>     if (testCase == "LibraryInitializerTest") {
>         const config = getConfig();
>         const exports = await getAssemblyExports(config.mainAssemblyName);
> 
>         exports.LibraryInitializerTest.Run();
>     }
>   }

Reviewer: ```suggestion
  export function onRuntimeReady({ getAssemblyExports, getConfig }) {
        // Sample: After the runtime starts, but before Main method is called, call [JSExport]ed method
        const config = getConfig();
        const exports = await getAssemblyExports(config.mainAssemblyName);
        exports.Sample.Greet();
  }
```

  ```

> 

Reviewer: ```suggestion
Both callbacks can return a `Promise` and the promise will be awaited before the startup will continue.
```
